esclient/esclient.go

```diff
@@ -0,0 +1,124 @@
+package esclient
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	dpAwsauth "github.com/ONSdigital/dp-elasticsearch/v2/awsauth"
+	dphttp "github.com/ONSdigital/dp-net/http"
+	"github.com/ONSdigital/dp-search-data-importer/config"
+	"github.com/ONSdigital/log.go/v2/log"
+
+	"github.com/pkg/errors"
+)
+
+const applicationJSON = "application/json"
+
+//go:generate moq -out mock/Client.go -pkg mock . Client
+
+// Client provides an interface with which to communicate with Elastic Search by way of HTTP requests
+type Client interface {
+	SubmitBulkToES(ctx context.Context, esDestIndex string, esDestURL string, bulk []byte) ([]byte, error)
+}
+
+// ClientImpl represents an instance of the elasticsearch client
+type ClientImpl struct {
+	client    dphttp.Clienter
+	requester Requester
+}
+
+// NewClient returns a concrete implementation of the Client interface
+func NewClient(client dphttp.Clienter) Client {
+
+	return &ClientImpl{
+		client:    client,
+		requester: NewRequester(),
+	}
+}
+
+// NewClientWithRequester returns a concrete implementation of the Client interface, taking a custom Requester
+func NewClientWithRequester(client dphttp.Clienter, requester Requester) Client {
+
+	return &ClientImpl{
+		client:    client,
+		requester: requester,
+	}
+}
+
+// SubmitBulkToES uses an HTTP post request to submit data to Elastic Search
+func (cli *ClientImpl) SubmitBulkToES(
```

Comment:
Why is this not in dp-elasticsearch library?

Reply:
BulkUpdate - new method name